packages/puppeteer-core/src/bidi/Page.ts

```diff
@@ -846,6 +846,51 @@ export class BidiPage extends Page {
     });
   }
 
+  override async cookies(): Promise<Protocol.Network.Cookie[]> {
+    const bidiCookies = await this.#connection.send('storage.getCookies', {
+      partition: this.mainFrame()._id,
+    });
+    return bidiCookies.result.cookies.map(c => {
+      return {
+        name: c.name,
+        // Present binary value as base64 string.
+        value: c.value.value,
+        domain: c.domain,
+        path: c.path,
+        size: c.size,
+        httpOnly: c.httpOnly,
+        secure: c.secure,
+        sameSite: this.#convertCookiesSameSite(c.sameSite),
+        // TODO: check default `expiry` value.
+        expires: c.expiry ?? -1,
+        // TODO: add proper value.
+        session: c.expiry === undefined || c.expiry <= 0,
+        // TODO: add proper value.
+        priority: 'Medium',
+        // TODO: add proper value.
+        sameParty: false,
+        // TODO: add proper value.
+        sourceScheme: 'NonSecure',
+        // TODO: add proper value.
+        sourcePort: -1,
+        // TODO: respect other `PartitionKey` values.
+        partitionKey: bidiCookies.result.partitionKey.sourceOrigin,
```

Comment:
`partitionKey` in Puppeteer is a string, but in BiDi it can contain a `UserContext` (incognito context) as well, and it is extendable by vendor-specific keys.
What is the expected behavior in such cases?

Reply:
I think the partitionKey in Puppeteer is always UserContext-specific. So we don't need to include the user context here. What we need is to filter out cookies that do not belong to the current page's context.

Reply:
It looks like it's not possible to get such cookies so it does not look like any action is needed. Eventually, we need to change the Puppeteer API to support partitioned cookies across the browsers better.